### DISCUSSION TITLE
Virtual Scrolling + Lazy Loading

### DISCUSSION THREAD

**Discussion by fotisskal** [primefaces/community#3120] [PrimeVue]
Hello fellas!

Please help me with the following.

I am trying to implement a DataTable component (version 4) with vertical virtual scrolling plus lazy loading.
I have a backend API call which fetches paginated data, eg. 10 per page.
The documentation has confused me a lot regarding this part.

Example virtualScrollOptions I pass to my DataTable:
`:virtualScrollerOptions="{ lazy: true, autoSize: true, showLoader: true, itemSize: 75, delay: 250, loading: lazyLoading, onLazyLoad: loadLazy, numToleratedItems: 10 }"`

The problem is the 1st 10 items are loaded but scrolling down does not trigger the onLazyLoad method.
Also event.first property seems always 0.

On documentation page they try to simulate this with timeouts but it is a bit unclear how to handle in a real scenario with API calls.
For example do I need to prefill the DataTable with a big items array?
Does itemSize play a role in this? And if yes, what if the row height is not always the same?

I would appreciate any help!

**Comment by jayel74** [comment]
Hello, I have the same issue? Was this problem solved? If yes, what is the solution please?

**Comment by jayel74** [comment]
Hello

Is there nobody that have an example how to do the above please?
